### PR TITLE
[Button] Support for small buttons

### PR DESCRIPTION
**Pull request by llorca**
#### Checklist

- [x] Update documentation

#### Changes proposed in this pull request:

Initial implementation of long requested small buttons, height 24px. No icon is acceptable at less than 16px, and we can't swap icon sets yet, therefore we `display: none` all icons within small buttons

#### Screenshot

image

### CODE REVIEW

File: packages/core/src/components/button/_button.scss [hunk 1]
```diff
@@ -238,6 +239,15 @@ Styleguide components.button.css
     }
   }
 
+  &.pt-small {
+    @include pt-button-small();
+
+    &[class*="pt-icon-"]::before,
+    #{$icon-classes} {
+      display: none;
```
Comment (giladgray) [inline packages/core/src/components/button/_button.scss]: definitely put a comment here about why we hide icons in small buttons.

File: packages/core/src/components/button/_button.scss [hunk 2]
```diff
@@ -41,6 +41,7 @@ Markup:
 .pt-intent-danger - Danger intent
 .pt-active - Active appearance
 .pt-large - Larger size
+.pt-small - Smaller size
```
Comment (giladgray) [inline packages/core/src/components/button/_button.scss]: "Smaller size. Note that small buttons do not support icons."

**Comment by pkwi** [comment]
+1 for the better explanation on why icons are hidden.

In Button groups section, .pt-small buttons still show as default.

**Comment by llorca** [comment]
Good catch, I'll fix the button groups

**Pull request merged**